gui/widgets/bandview.cpp

```diff
@@ -41,9 +46,41 @@ void BandView::initUi()
 			this, SLOT(commitLabelChanges()));
 }
 
+
+void BandView::updateMode(SelectionMode mode)
+{
+    sm = mode;
+
+}
+
+
+void BandView::resizeEvent()
+{
+    ScaledView::resizeEvent();
+    zoom = 1;
+}
+
+void BandView::wheelEvent(QGraphicsSceneWheelEvent *event)
+{
+
+    ScaledView::wheelEvent(event);
+
+}
+
+
 void BandView::setPixmap(QPixmap p)
 {
-	ScaledView::setPixmap(p);
+    //ScaledView::setPixmap(p);
```

Comment:
This code does nothing else than the one in scaledView, except being more clever. Why not still call ScaledView's method here but in there, incorporate the condition on resizeEvent and updateSizeHint.